<?xml version="1.0"?>
<launch>

    <!-- Board localization rguments -->
    <!-- <arg name="camera_name" default="camera"/> -->
    <arg name="camera_name" default="d435_camera"/>
    <arg name="camera_info_topic_name" default="/$(arg camera_name)/color/camera_info"/>
    <arg name="image_raw_topic_name" default="/$(arg camera_name)/color/image_raw"/>
    <arg name="reference_frame" default="world"/>
    <arg name="board_localization_node_name" default="board_detection_fsm"/>
    <arg name="sim" default="false"/>
    
    <arg unless="$(arg sim)" name="dyn_rec_config_file" default="$(find hrii_board_localization)/config/board_localization_params_real.yaml"/> 
    <arg if="$(arg sim)" name="dyn_rec_config_file" default="$(find hrii_board_localization)/config/board_localization_params_sim.yaml"/> 

    <group ns="fsm">
        <node pkg="hrii_board_localization" type="board_localization_fsm.py" name="$(arg board_localization_node_name)" output="screen" required="true">
            <param name="camera_info_topic_name" value="$(arg camera_info_topic_name)"/>
            <param name="image_raw_topic_name" value="$(arg image_raw_topic_name)"/>
            <param name="reference_frame" value="$(arg reference_frame)"/>
        </node>

        <node pkg="dynamic_reconfigure" type="dynparam" name="dynamic_reconfigure_load" args="load $(arg board_localization_node_name) $(arg dyn_rec_config_file)"/>
    </group>

</launch>